src/FEValues/common_values.jl

```diff
@@ -172,6 +202,16 @@ function function_gradient(fe_v::AbstractValues, q_point::Int, u::AbstractVector
     end
     return grad
 end
+function_gradient(iv::InterfaceValues, q_point::Int, u::AbstractVector, dof_range = eachindex(u); here::Bool = true) = begin
+    fv = here ? iv.face_values : iv.other_face_values
+    if dof_range != eachindex(u)
+        here && any(dof_range .> length(eachindex(u)) ÷ 2) && error("dof_range contains dof on the other face of the interface")
+        here || any(dof_range .<= length(eachindex(u)) ÷ 2) && error("dof_range contains dof on the other face of the interface")
+    else
+        dof_range = here ? dof_range[1 : end ÷ 2] : dof_range[end ÷ 2 + 1 : end]
+    end
```

Comment:
I think we might want to revisit this logic here.

Reply:
If I understand correctly, `u` is passed as the dof from both elements on the interface, so this part makes sure if we pass `dof_range` it does not contain dof from the other element, and in case no `dof_range` is passed it takes the first/second half depending on which element we use.
I see this may be problematic in case of continuous interpolations, is this what you meant?

Reply:
Yes, this is on example. I had no time to do a full audit of the code, so I might misunderstand something fundamental here, but you introduce the input argument `dof_range` which defaults to `eachindex(u)`. Now these lines here exactly check for `dof_range == eachindex(u)`, which raises for me the question why `dof_range` has been introduced as an input argument in the first place. Can you elaborate a bit on the design here?

Reply:
I'm not sure why it is there as it was there for `FaceValues` implementation.